pkg/monitortests/network/onpremkeepalived/keepalived_priority_check.go

```diff
@@ -0,0 +1,265 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"regexp"
+	"sort"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortestlibrary/podaccess"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type operatorLogAnalyzer struct {
+	kubeClient kubernetes.Interface
+}
+
+func InitialAndFinalOperatorLogScraper() monitortestframework.MonitorTest {
+	return &operatorLogAnalyzer{}
+}
+
+func (w *operatorLogAnalyzer) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	var err error
+	w.kubeClient, err = kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := scanAllOperatorPods(ctx, w.kubeClient, newOperatorLogHandler(recorder)); err != nil {
+		return fmt.Errorf("unable to scan operator logs: %w", err)
+	}
+
+	return nil
+}
+
+func scanAllOperatorPods(ctx context.Context, kubeClient kubernetes.Interface, logHandlers ...podaccess.LogHandler) error {
+	onPremPlatforms := []string{"kni", "openstack", "vsphere"}
+	errs := []error{}
+	for _, platform := range onPremPlatforms {
+
+		pods, err := kubeClient.CoreV1().Pods(fmt.Sprintf("openshift-%s-infra", platform)).List(ctx, metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s-infra-vrrp", platform)})
+		if err != nil {
+			return fmt.Errorf("couldn't list pods: %w", err)
+		}
+
+		for _, pod := range pods.Items {
+			// this is just a basic check to see if we can expect logs to be present. Unready, unhealthy, and failed pods all still have logs.
+			if pod.Status.Phase == corev1.PodPending || pod.Status.Phase == corev1.PodUnknown {
+				continue
+			}
+
+			for _, container := range pod.Spec.Containers {
+				if container.Name == "keepalived" {
+					streamer := podaccess.NewOneTimePodStreamer(kubeClient, pod.Namespace, pod.Name, container.Name, logHandlers...)
+					if err := streamer.ReadLog(ctx); err != nil && !apierrors.IsNotFound(err) {
+						errs = append(errs, fmt.Errorf("error reading log for pods/%s -n %s -c %s: %w", pod.Name, pod.Namespace, container.Name, err))
+					}
+				}
+			}
+		}
+	}
+	return errors.Join(errs...)
+}
+
+func (w *operatorLogAnalyzer) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	localRecorder := monitor.NewRecorder()
+	if err := scanAllOperatorPods(ctx, w.kubeClient, newOperatorLogHandlerAfterTime(localRecorder, beginning)); err != nil {
+		return nil, nil, fmt.Errorf("unable to scan operator logs: %w", err)
+	}
+
+	return localRecorder.Intervals(time.Time{}, time.Time{}), nil, nil
+}
+
+func (*operatorLogAnalyzer) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
```

Comment:
Do you need to make use of beginning and end to prevent picking up events outside the current execution? In an upgrade job openshift-tests is invoked twice, once for the upgrade and once for conformance after. You need to bound the creation of intervals to only those which fit within this execution otherwise the conformance graphs will blow out to include all of upgrade. I didn't see any on-oprem lb upgrade jobs run but you could kick one off with /payload-job.

If it's actually in here and I've missed it then my apologies.